content/blog/2023/2023-03-31-serialize-deserialize-with-jsonview.md

```diff
@@ -0,0 +1,617 @@
+{{% image alt="settings" src="images/posts/spring-boot-jsonview/MapperFeature.JPG" %}}
+
+We can also verify the configuration defaults when the `JacksonAutoConfiguration` class gets registered using the below test:
+````java
+@SpringBootTest
+public class JacksonAutoConfigTest {
+
+    private AnnotationConfigApplicationContext context;
+
+    @Test
+    public void defaultObjectMapperBuilder() throws Exception {
+        this.context.register(JacksonAutoConfiguration.class);
+        this.context.refresh();
+        Jackson2ObjectMapperBuilder builder = 
+                this.context.getBean(Jackson2ObjectMapperBuilder.class);
+        ObjectMapper mapper = builder.build();
+        assertTrue(MapperFeature.DEFAULT_VIEW_INCLUSION.enabledByDefault());
+        assertFalse(mapper.getDeserializationConfig().isEnabled(
+                MapperFeature.DEFAULT_VIEW_INCLUSION));
+        assertFalse(mapper.getSerializationConfig().isEnabled(
+                MapperFeature.DEFAULT_VIEW_INCLUSION));
+    }
+}
+````
+`SerializationConfig` and `DeserializationConfig` classes contain baseline configurations for serialization and deserialization processes respectively.
+**When the ObjectMapper's `MapperFeature.DEFAULT_VIEW_INCLUSION` value is enabled, it automatically applies to both serialization and deserialization process.**
+We can override this value to apply to either serialization only or deserialization only (if required by enabling or disabling the MapperFeature configuration) using `SerializationConfig` and `DeserializationConfig` classes respectively.
+We will take a look at its configuration in the further sections.
+
+## Using @JsonView with Spring Boot
+Let's look at a sample User application to demonstrate the various ways in which @JsonView annotation can be used.
+This application is configured to run on port 8083.
+````text
+mvnw clean verify spring-boot:run (for Windows)
+./mvnw clean verify spring-boot:run (for Linux)
+````
+
+Let's first define the views.
+````java
+package com.reflectoring.userdetails.persistence;
+
+public class Views {
+    // For external user
+    public static interface ExternalView {
+
+    }
+    // For internal user
+    public static interface InternalView extends ExternalView {
+
+    }
+    // Basic User Details
+    public static interface UserSummary {
+
+    }
+    // Additional User Details
+    public static interface UserDetailedSummary extends UserSummary {
+
+    }
+    // Default fields mapped for GET requests
+    public static interface GetView {
+
+    }
+    // Allowed fields for PATCH requests
+    public static interface PatchView {
+
+    }
+}
+````
+In the further sections, we will take a look at how to use `@JsonView` to cater to each of the usecases we previously looked at.
+
+{{% info title="Difference between @JsonView and @JsonIgnore" %}}
```

Comment:
It would be better to include the explanation of the difference in the section above where you introduce the @JsonView annotation. This would help provide a more comprehensive understanding of how to use @JsonView effectively.